src/extensions/default/JavaScriptCodeHints/main.js

```diff
@@ -56,6 +56,9 @@ define(function (require, exports, module) {
         noHintsOnDot   = false, // preference setting to prevent hints on dot
         ignoreChange;           // can ignore next "change" event if true;
 
+    // Languages that support inline JavaScript
+    var _inlineScriptLanguages = ["html", "php"];
```

Comment:
Maybe is something that could be defined as part of the language in the LanguageManager?